tests/ci/gtest_util.sh

```diff
@@ -0,0 +1,26 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+
+function shard_gtest() {
+    export GTEST_TOTAL_SHARDS=$(nproc --all)
+    if [ -n "${2}" ]; then
+        GTEST_TOTAL_SHARDS="${2}"
+    fi
+    if [ -z ${GTEST_TOTAL_SHARDS} -o ${GTEST_TOTAL_SHARDS} -lt 1 ]; then
+        GTEST_TOTAL_SHARDS=4
+    fi
+
+    PIDS=()
+    COUNTER=0
+    while [ $COUNTER -lt $GTEST_TOTAL_SHARDS ]; do
+        export GTEST_SHARD_INDEX=$COUNTER
+        echo shard_gtest-Command: ${1}
+        ${1} &
+        PIDS[${COUNTER}]=$!
+        COUNTER=$(( COUNTER+1 ))
+    done
+
+    for PID in ${PIDS[*]}; do
+        wait -f $PID
```

Comment:
will this correctly propagate each process's exit code? we wouldn't want to let an individual shard silently fail in our CI...

Reply:
I was assuming that our scripts all use `set -e`, but that's not a safe assumption.  I'll modify this to explicitly check the exit status of the child and return an appropriate value.